src/components/NewCommentForm.tsx

```
@@ -1,8 +1,70 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+
+type Props = {
+  openedPost: Post;
+  loading: boolean;
+  addCommentHandler: (
+    postId: number,
+    name: string,
+    email: string,
+    body: string,
+  ) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  openedPost,
+  loading,
+  addCommentHandler,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [commentError, setCommentError] = useState(false);
+
+  const clearAllHandler = () => {
+    setName('');
+    setEmail('');
+    setCommentText('');
+    setNameError(false);
+    setEmailError(false);
+    setCommentError(false);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const normalizedName = name.trim();
+    const normalizedEmail = email.trim();
+    const normalizedComment = commentText.trim();
+
+    if (!normalizedName) {
+      setNameError(true);
+    }
+
+    if (!normalizedEmail) {
+      setEmailError(true);
+    }
+
+    if (!normalizedComment) {
+      setCommentError(true);
+    }
+
+    if (normalizedName && normalizedEmail && normalizedComment) {
+      try {
+        await addCommentHandler(openedPost.id, name, email, commentText);
+      } catch {
```

Empty catch block in `handleSubmit`. The handler swallows errors silently (`catch { }`). Either surface an error to the user or rethrow so the caller can handle it. At minimum, add a comment explaining why it's intentionally empty if that was deliberate.